src/refer.ts

```diff
@@ -81,32 +100,44 @@ function refer(this: any, options: any) {
     }
   }
 
-
   async function prepare(this: any) {
-    let seneca = this
+    const seneca = this
     await seneca.post('biz:refer,load:rules')
   }
-
 }
 
+async function getEntryReward(seneca: any, entry: any): Promise<object> {
+  const reward = await seneca.entity('refer/reward').load$({
+    entry_id: entry.id
+  })
+
+  if (!reward) {
+    return await seneca.entity('refer/reward').save$({
```

Comment:
let's make this a little more DRY: create the entity, but do not save, and set count to 0. Do not return, drop through to lines 123,124 to do the save

Reply:
Thank you for the suggestion. I updated this script, please take a look again.